src/main/java/net/fabricmc/loom/task/GenVsCodeProjectTask.java

```diff
@@ -140,7 +139,7 @@ public void genRuns() throws IOException {
 			toRemove.forEach(configurations::remove);
 			configurations.add(configurationJson);
 
-			Files.createDirectories(Paths.get(configuration.runDir));
+			Files.createDirectories(configuration.absoluteRunDirPath(getProject()));
```

Comment:
You cannot use `getProject()` during task execution, this is casusing the tests to fail. I might have a look at this.